usaspending_api/awards/v2/data_layer/orm.py

```diff
@@ -577,3 +585,91 @@ def fetch_naics_hierarchy(naics: str) -> dict:
+            distinct concat(p.reporting_fiscal_year::text,
+            lpad(p.reporting_fiscal_period::text, 2, '0')) as fyp,
+            p.reporting_fiscal_year, p.reporting_fiscal_period
+        from submission_attributes p
+        where now()::date > p.reporting_period_end -- change "period end" with "window close"
+        order by p.reporting_fiscal_year desc, p.reporting_fiscal_period desc
+    ),
+    fy_final_balances as (
+        -- Rule: If a balance is not zero at the end of the year, it must be reported in the
+        -- final period's submission (month or quarter), otherwise assume it to be zero
+        select faba.award_id, sum(faba.gross_outlay_amount_by_award_cpe) as prior_fys_outlay,
+            faba.disaster_emergency_fund_code
+        from eligible_file_c_records faba
+        group by
+            faba.award_id,
+            faba.reporting_fiscal_period,
+            faba.disaster_emergency_fund_code
+        having faba.reporting_fiscal_period = 12
+        and sum(faba.gross_outlay_amount_by_award_cpe) > 0
+    ),
+    current_fy_balance as (
+        select
+            faba.award_id,
+            faba.reporting_fiscal_year,
+            faba.reporting_fiscal_period,
+            faba.disaster_emergency_fund_code,
+            sum(faba.gross_outlay_amount_by_award_cpe) as current_fy_outlay
+        from eligible_file_c_records faba
+        group by
+            faba.award_id,
+            faba.reporting_fiscal_year,
+            faba.reporting_fiscal_period,
+            faba.disaster_emergency_fund_code
+        having concat(faba.reporting_fiscal_year::text, lpad(faba.reporting_fiscal_period::text, 2, '0')) in
+        (select max(fyp) from closed_periods) and sum(faba.gross_outlay_amount_by_award_cpe) > 0)
+    select
+        faba.disaster_emergency_fund_code,
+        coalesce(ffy.prior_fys_outlay, 0) + coalesce(cfy.current_fy_outlay, 0) as total_outlay,
+        coalesce(sum(faba.transaction_obligated_amount), 0) as obligated_amount
+    from eligible_file_c_records faba
+    left join fy_final_balances ffy on ffy.award_id = faba.award_id and ffy.disaster_emergency_fund_code = faba.disaster_emergency_fund_code
+    left join current_fy_balance cfy
+        on cfy.reporting_fiscal_period != 12 -- don't duplicate the year-end period's value if in unclosed period 01
+        and cfy.award_id = faba.award_id
+        and cfy.disaster_emergency_fund_code = faba.disaster_emergency_fund_code
+    group by faba.disaster_emergency_fund_code, total_outlay;
+    """
+
+
+def fetch_account_details_award(award_id: int) -> dict:
+    award_id_sql = "faba.award_id = {award_id}".format(award_id=award_id)
+    results = execute_sql_to_ordered_dictionary(defc_sql.format(award_id_sql=award_id_sql))
+    outlay_by_code = []
+    obligation_by_code = []
+    total_outlay = 0
+    total_obligations = 0
+    for row in results:
+        total_outlay += row["total_outlay"]
+        total_obligations += row["obligated_amount"]
```

Comment:
Here and below, this is going to give bad sums. 

Looks like you removed this part from the sql:
```
inner join disaster_emergency_fund_code defc on defc.group_name = 'covid_19' and defc.code = faba.disaster_emergency_fund_code
```

Which was necessary to get the breakdowns by ALL DEFCs for the award details. However, the total covid obligation and total covid outlay should be just based on DEFC rows that have `group_name = 'covid_19'`. 

So probably run a query on the defc reference table before running this query to get the covid_19 defcs, and then in this row only do the sum if the row's defc is one of those in the group.

This is based on the assumption that we want ALL DEFC funding breakdowns eventually, and not just the covid ones. If we're only working with covid ones ... add that inner join back to the CTE for eligible file c records.

Reply:
It would be great to also see another unit test that creates file c rows with DEFCs that are and are _not_ COVID-19 defcs, and assert that the totals are only inclusive of the ones that are covid DEFCs. 

Also once the closed periods stuff is in, a similar test would be good for excluding outlay balances that are either from non-period-12 in prior FYs or from not-the-latest-closed-period-in-this-FY.

Reply:
So as part of my testing I wrote up a bunch of additional tests.  Once I'm done helping Melissa, I'll pass them along to Emily.  They added non COVID DEF codes and a bunch of other things.

Reply:
@emily-brents - I think this is still an issue with the new SQL. The new SQL doesn't limit faba records to just those with `group_name = 'covid_19'` so this looping and summing will sum across all/any DEFCs submitted (including Q and 9, etc. that are not COVID-related). 

Whereas the AC for this effort is to show total outlay and total obligation for just COVID-related def codes. 

You need some condition in the loop that says something like:
```
covid_defcs = DisasterEmergencyFundCodes.objects().filter("group_name = 'covid_19'").values_list('code', flat=True)
# ...
for row in results:
    if row["disaster_emergency_fund_code"] in covid_defcs:
        total_outlay += row["total_outlay"]
        total_obligations += row["obligated_amount"]
# ...
```